src/Test/L1/Mock/FakeTaskManager.cs

```diff
@@ -0,0 +1,29 @@
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System;
+using System.Linq;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using Pipelines = Microsoft.TeamFoundation.DistributedTask.Pipelines;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.L1.Worker
+{
+    public class FakeTaskManager : TaskManager
+    {
+        public override Definition Load(Pipelines.TaskStep task)
+        {
+            Definition d = base.Load(task);
+            if (task.Reference.Id == Pipelines.PipelineConstants.CheckoutTask.Id && task.Reference.Version == Pipelines.PipelineConstants.CheckoutTask.Version)
```

Comment:
We need this because checkout gets special cased - https://github.com/microsoft/azure-pipelines-agent/blob/615ab15a70c632397e35dd56cccba0aecd550878/src/Agent.Worker/TaskManager.cs#L85